src/plugins/auth.js

```diff
@@ -0,0 +1,47 @@
+"use strict";
+
+const log = require("../log");
+const colors = require("chalk");
+
+// The order defines priority: the first available plugin is used.
+// Always keep 'local' auth plugin at the end of the list; it should always be enabled.
+const plugins = [require("./auth/ldap"), require("./auth/local")];
+
+function unimplemented(funcName) {
+	log.debug(
+		`Auth module ${colors.bold(
+			module.exports.moduleName
+		)} doesn't implement function ${colors.bold(funcName)}`
+	);
+}
+
+// Default API implementations
+module.exports = {
+	moduleName: "<module with no name>",
+
+	// Must override: implements authentication mechanism
+	auth: () => unimplemented("auth"),
+
+	// Optional to override: implements filter for loading users at start up
+	// This allows an auth plugin to check if a user is still acceptable, if the plugin
+	// can do so without access to the user's unhashed password.
+	// Returning 'false' triggers fallback to default behaviour of loading all users
+	loadUsers: () => false,
+};
+
+// Override default API stubs with exports from first enabled plugin found
+for (let i = 0; i < plugins.length; ++i) {
+	if (plugins[i].isEnabled()) {
+		let name;
+
+		for (name in plugins[i]) {
+			module.exports[name] = plugins[i][name];
+		}
+
+		break;
+	}
+}
+
+if (module.exports.length === 0) {
```

Comment:
This won't happen as you're exporting some defaults.

Reply:
Oops. Thanks.